lib/origen_testers/igxl_based_tester/base/edgesets.rb

```diff
@@ -0,0 +1,53 @@
+module OrigenTesters
+  module IGXLBasedTester
+    class Base
+      class Edgesets
+        include ::OrigenTesters::Generator
+
+        attr_accessor :es
+
+        OUTPUT_PREFIX = 'ES_'
+        OUTPUT_POSTFIX = ''
+
+        def initialize # :nodoc:
+          @es = {}
+        end
+
+        def add(esname, pin, edge, options = {})
+          esname = esname.to_sym unless esname.is_a? Symbol
+          pin = pin.to_sym unless pin.is_a? Symbol
+          @es.key?(esname) ? @es[esname].add_edge(pin, edge) : @es[esname] = platform::Edgeset.new(esname, pin, edge, options)
+          @es
+        end
+
+        def finalize(options = {})
+        end
+
+        # Equality check to compare full contents of edge object
+        def edges_eql?(edge1, edge2)
+          edge1.d_src == edge2.d_src && edge1.d_fmt == edge2.d_fmt &&
+            edge1.d0_edge == edge2.d0_edge && edge1.d1_edge == edge2.d1_edge &&
+            edge1.d2_edge == edge2.d2_edge && edge1.d3_edge == edge2.d3_edge &&
+            edge1.c_mode == edge2.c_mode &&
+            edge1.c1_edge == edge2.c1_edge && edge1.c2_edge == edge2.c2_edge &&
+            edge1.t_res == edge2.t_res && edge1.clk_per == edge2.clk_per
+        end
```

Comment:
@aaronburg, if you implement the == method it won't care about object ID unless you include it in your comparison. In fact the whole point of doing that is precisely because you still want to be able to consider two different objects equal if the attributes you are about are the same.